File: wiz_dashboard/domain/formatting.py
"""Human-friendly duration and size formatting."""

import pandas as pd


def format_bytes(n) -> str:
    """Human-friendly byte size ("1.2 MB"). None/negative -> "0 B"."""
    n = 0 if n is None else max(0, float(n))
    for unit in ("B", "KB", "MB", "GB"):
        if n < 1024 or unit == "GB":
            return f"{int(n)} {unit}" if unit == "B" else f"{n:.1f} {unit}"
        n /= 1024
    return f"{n:.1f} GB"


def format_duration(days):
    """Human-friendly duration. None/NaN -> em-dash."""
    if days is None or pd.isna(days):
        return "—"
    if days < 1 / 24:
        return "<1h"
    if days < 1:
        return f"{int(round(days*24))}h"
    if days < 30:
        return f"{days:.1f}d"
    if days < 365:
        return f"{days/30:.1f}mo"
    return f"{days/365:.1f}y"

File: gas/src/domain/formatting.ts
// Human-friendly duration and size formatting — the port of domain/formatting.py.

export function formatBytes(n: number | null | undefined): string {
  let v = n === null || n === undefined ? 0 : Math.max(0, Number(n));
  for (const unit of ["B", "KB", "MB", "GB"] as const) {
    if (v < 1024 || unit === "GB") {
      return unit === "B" ? `${Math.trunc(v)} ${unit}` : `${v.toFixed(1)} ${unit}`;
    }
    v /= 1024;
  }
  return `${v.toFixed(1)} GB`;
}

export function formatDuration(days: number | null | undefined): string {
  if (days === null || days === undefined || Number.isNaN(days)) return "—";
  if (days < 1 / 24) return "<1h";
  if (days < 1) return `${Math.round(days * 24)}h`;
  if (days < 30) return `${days.toFixed(1)}d`;
  if (days < 365) return `${(days / 30).toFixed(1)}mo`;
  return `${(days / 365).toFixed(1)}y`;
}
